### ISSUE TITLE
Add non-thread safe version

### ISSUE THREAD

**Issue by madsmtm** [drewcrawford/continue#2]
Would probably make sense for thread-local executors, especially given that AppKit only runs on the main thread, so if one were to make an executor based on NSRunLoop etc., you'd probably want.

Not entirely sure though, haven't actually implemented such an executor yet.

This would be more performant (both because it could use `LocalWaker` once stable, and because it could use `Cell`/`Rc` instead of atomics and `Arc`) and the implementation would probably be simpler.

**Comment by drewcrawford** [comment]
I agree this makes sense to do.  It is not high on my own todo list because uncontested atomic writes are fast enough that I rarely need to think about them, relative to the cost of switching to the "other" tool when I discover I needed it after all.  But of course there are usecases.

The one you mention, I am working on a single-threaded executor right now.  So far, I have not felt the need for a single-threaded continuation type over there, but perhaps I will by the time I get it working.
